gitoxide-core/src/repository/diff.rs

```diff
@@ -0,0 +1,92 @@
+use gix::bstr::{BString, ByteSlice};
+
+pub fn tree(
+    repo: gix::Repository,
+    out: &mut dyn std::io::Write,
+    old_treeish: BString,
+    new_treeish: BString,
+) -> anyhow::Result<()> {
+    let old_tree_id = repo.rev_parse_single(old_treeish.as_bstr())?;
+    let new_tree_id = repo.rev_parse_single(new_treeish.as_bstr())?;
+
+    let old_tree = old_tree_id
+        .object()
+        .unwrap()
+        .peel_to_kind(gix::object::Kind::Tree)
+        .unwrap()
+        .into_tree();
+    let new_tree = new_tree_id
+        .object()
+        .unwrap()
+        .peel_to_kind(gix::object::Kind::Tree)
+        .unwrap()
+        .into_tree();
+
+    let changes = repo.diff_tree_to_tree(&old_tree, &new_tree, None)?;
```

Comment:
Nice to see you found this (new) method useful.

Reply:
It was when I saw it in the history that I decided to open this PR. :-)